src/blocks/memory.rs

```diff
@@ -139,17 +130,10 @@ pub async fn run(config: toml::Value, mut api: CommonApi) -> Result<()> {
             "cached_percent" => Value::percents(cached / mem_total * 100.)
         });
 
-        widget.state = match memtype {
-            Memtype::Memory => match mem_used / mem_total * 100. {
-                x if x > config.critical_mem => State::Critical,
-                x if x > config.warning_mem => State::Warning,
-                _ => State::Idle,
-            },
-            Memtype::Swap => match swap_used / swap_total * 100. {
-                x if x > config.critical_swap => State::Critical,
-                x if x > config.warning_swap => State::Warning,
-                _ => State::Idle,
-            },
+        widget.state = match mem_used / mem_total * 100. {          
+                x if x > config.critical_mem || x > config.critical_swap => State::Critical,
+                x if x > config.warning_mem || x > config.warning_swap => State::Warning,
+                _ => State::Idle,           
         };
```

Comment:
`mem_used / mem_total` should be compared against `critical_mem` and `warning_mem`, while `swap_used / swap_total` should be compared against `critical_swap` and `warning` swap. If at lest one of them is "critical", then the block should have "critical" state. Same for "warning".